py/aou_workbench_client/cohorts.py

```diff
@@ -1,12 +1,46 @@
 from aou_workbench_client.auth import get_authenticated_swagger_client
 from aou_workbench_client.config import all_of_us_config
 from aou_workbench_client.swagger_client.apis.cohorts_api import CohortsApi
+from copy import deepcopy
 
-def materialize_cohort(request):
-    """Materializes a cohort in the workspace containing this notebook, based
-    on the provided MateralizedCohortRequest."""
+def materialize_cohort_page(request):
+    """Returns a MaterializeCohortResponse representing a page of results 
+    from materializing a cohort in the workspace containing this notebook, based
+    on the provided MateralizeCohortRequest."""
     client = get_authenticated_swagger_client()
     cohorts_api = CohortsApi(api_client=client)
     return cohorts_api.materialize_cohort(all_of_us_config.workspace_namespace,
                                           all_of_us_config.workspace_id,
                                           request=request)
+
+def materialize_cohort(request, max_results=None):
+    """Materializes a cohort in the workspace containing this notebook, based
+    on the provided MateralizeCohortRequest. Returns a generator of 
+    dictionaries containing the results.
+    
+    If max_results is specified, up to that many results will be returned; 
+    if not, all available results will be returned. 
+    
+    Multiple server requests may be made to retrieve all the results,
+    using the page_size specified in the request for each request."""
+    client = get_authenticated_swagger_client()
+    cohorts_api = CohortsApi(api_client=client)
+    num_results = 0
+    # Clone the request, since we're going to be modifying it.
+    request = deepcopy(request)    
+    page_size = request.page_size or 1000
```

Comment:
Does the server not pick a default?

Reply:
It does, but I can't adjust the page size appropriately below to not get more results than we need if I don't know what the default page size is going to be. So I had to duplicate it here.

If the server default page size changes, this will still work... it just won't use the server default.